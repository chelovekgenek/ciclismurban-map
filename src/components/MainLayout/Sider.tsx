import React, { useCallback } from "react"
import { connect } from "react-redux"

import { Layout, Checkbox, Grid } from "components/generic/ui"
import { TAppState } from "store/entities"
import { getRoot, toggle } from "store/entities/filters"

import * as Styled from "./Sider.styled"

const { Row } = Grid

interface IStateProps {
  filters: ReturnType<typeof getRoot>
}
interface IDispatchProps {
  toggle: typeof toggle
}

interface IProps extends IStateProps, IDispatchProps {}

export const Sider: React.FC<IProps> = ({ filters, toggle }) => {
  const toggleServices = useCallback(() => {
    toggle("services")
  }, [toggle])
  const toggleParkings = useCallback(() => {
    toggle("parkings")
  }, [toggle])
  const toggleShops = useCallback(() => {
    toggle("shops")
  }, [toggle])
  return (
    <Layout.Sider width="14rem" theme="light">
      <Styled.Col>
        <Row>
          <Checkbox checked={filters.services} onChange={toggleServices}>
            Services
          </Checkbox>
        </Row>
        <Row>
          <Checkbox checked={filters.parkings} onChange={toggleParkings}>
            Parkings
          </Checkbox>
        </Row>
        <Row>
          <Checkbox checked={filters.shops} onChange={toggleShops}>
            Shops
          </Checkbox>
        </Row>
      </Styled.Col>
    </Layout.Sider>
  )
}

export default connect<IStateProps, IDispatchProps, null, TAppState>(
  state => ({
    filters: getRoot(state),
  }),
  {
    toggle,
  },
)(Sider)
