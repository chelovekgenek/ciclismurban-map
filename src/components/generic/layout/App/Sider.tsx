import React, { useState, useMemo } from "react"
import { withRouter, RouteComponentProps, Link } from "react-router-dom"

import { Icon } from "components/generic/ui"

import options from "./Sider.options.json"
import * as Styled from "./Sider.styled"

interface IProps extends RouteComponentProps {}

export const Sider: React.FC<IProps> = ({ match }) => {
  const [collapsed, setCollapsed] = useState(false)
  const selected = useMemo(() => {
    for (let option of options) {
      if (option.link === match.path || (option.routes && option.routes.includes(match.path))) {
        return [option.link]
      }
    }
    return []
  }, [match.path])
  return (
    <Styled.Sider collapsible collapsed={collapsed} onCollapse={setCollapsed}>
      <Styled.Menu selectedKeys={selected}>
        {options.map(item => (
          <Styled.MenuItem key={item.link}>
            <Icon type={item.icon} />
            <Link to={item.link}>{item.title}</Link>
          </Styled.MenuItem>
        ))}
      </Styled.Menu>
    </Styled.Sider>
  )
}

export default withRouter(Sider)
