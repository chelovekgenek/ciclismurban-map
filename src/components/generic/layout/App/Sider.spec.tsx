import React from "react"
import { shallow } from "enzyme"

import { Sider } from "./Sider"
import * as Styles from "./Sider.styled"

describe("Sider", () => {
  const render = (props: Partial<React.ComponentProps<typeof Sider>> = {}) =>
    shallow(<Sider match={{ params: {} } as any} history={{} as any} location={{} as any} {...props} />)
  it("should match snapshot", () => {
    expect(render()).toMatchSnapshot()
  })
  // it("should redirect to link when item was clicled", () => {
  //   const props = { history: { push: jest.fn() } }
  //   const wrapper = render(props as any)
  //   const menuItem = wrapper.find(Styles.MenuItem).at(0)
  //   menuItem.simulate("click")
  //   expect(props.history.push).toBeCalledWith("/")
  // })
  it("should not set selected menu item if doesn't match with current route", () => {
    const wrapper = render({ match: { path: "/asd", isExact: true, url: "/asd", params: {} } })
    expect(wrapper.find(Styles.Menu).prop("selectedKeys")).toEqual([])
  })
  it("should not set selected menu item if it's matching with current route", () => {
    const wrapper = render({ match: { path: "/", isExact: true, url: "/", params: {} } })
    expect(wrapper.find(Styles.Menu).prop("selectedKeys")).toEqual(["/"])
  })
  it("should set selected menu item if current route is a children", () => {
    const wrapper = render({
      match: { path: "/event/:id", isExact: true, url: "/event/12345", params: { id: "12345" } },
    })
    expect(wrapper.find(Styles.Menu).prop("selectedKeys")).toEqual(["/events"])
  })
  it("should hide title if scrollbar is collapsed", () => {
    const wrapper = render()
    ;(wrapper.find(Styles.Sider).prop("onCollapse") as Function)(true)
    expect(wrapper).toMatchSnapshot()
  })
})
