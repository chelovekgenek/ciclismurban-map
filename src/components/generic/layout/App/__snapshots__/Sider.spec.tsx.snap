// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/ServiceForm should match snapshot 1`] = `
<Styled(Sider)
  collapsed={false}
  collapsible={true}
  onCollapse={[Function]}
>
  <Styled(Menu)
    selectedKeys={Array []}
  >
    <Styled(MenuItem)
      key="/"
    >
      <Icon
        type="environment"
      />
      <Link
        to="/"
      >
        Карта
      </Link>
    </Styled(MenuItem)>
    <Styled(MenuItem)
      key="/events"
    >
      <Icon
        type="compass"
      />
      <Link
        to="/events"
      >
        Cобытия
      </Link>
    </Styled(MenuItem)>
    <Styled(MenuItem)
      key="/parkings"
    >
      <Icon
        type="car"
      />
      <Link
        to="/parkings"
      >
        Парковки
      </Link>
    </Styled(MenuItem)>
    <Styled(MenuItem)
      key="/services"
    >
      <Icon
        type="setting"
      />
      <Link
        to="/services"
      >
        Сервисы
      </Link>
    </Styled(MenuItem)>
    <Styled(MenuItem)
      key="/shops"
    >
      <Icon
        type="shop"
      />
      <Link
        to="/shops"
      >
        Магазины
      </Link>
    </Styled(MenuItem)>
  </Styled(Menu)>
</Styled(Sider)>
`;
